### DISCUSSION TITLE
How to update workflow status from designer?

### DISCUSSION THREAD

**Discussion by rairafaqatali** [elsa-workflows/elsa-core#2409] [Q&A]
Hi @sfmskywalker 
Can you let me know how to update workflow status from idle to running and from running to complete?

Thanks

**Comment by sfmskywalker** [comment]
You mean manually by changing the status on the workflow instance directly, or by actually running the workflow? What's your scenario?

**Reply by rairafaqatali** [reply]
I need an activity that can update the workflow status. Based on event hub events, I have to update the workflow status.

**Reply by sfmskywalker** [reply]
I see. In that case, all you'd need is an activity that handles the event hub events to start/resume workflows. These activities themselves don't have to change the workflow status; this is handled by the workflow runner for you.

For example, let's say you have a workflow instance in Idle that starts with the `Some Event Hub Event Received` activity; this would change the workflow status from Idle to Running and then Complete.

If you have multiple such activities, the workflow status would change automatically from Idle -> Running -> Suspended -> Running -> Completed.

But if you for some reason unbeknownst to me do need to change the workflow status explicitly, you can do so directly by changing the `WorkflowStatus` property of the `WorkflowInstance` itself.
